go-controller/pkg/node/node.go

```diff
@@ -595,6 +604,95 @@ func (n *OvnNode) WatchEndpoints() error {
 	return err
 }
 
+func exGatewayPodsAnnotationsChanged(oldNs, newNs *kapi.Namespace) bool {
+	// In reality we only care about exgw pod deletions, however since the list of IPs is not expected to change
+	// that often, let's check for *any* changes to these annotations compared to their previous state and trigger
+	// the logic for checking if we need to delete any conntrack entries
+	return (oldNs.Annotations[util.ExternalGatewayPodIPsAnnotation] != newNs.Annotations[util.ExternalGatewayPodIPsAnnotation]) ||
+		(oldNs.Annotations[util.RoutingExternalGWsAnnotation] != newNs.Annotations[util.RoutingExternalGWsAnnotation])
+}
+
+func (n *OvnNode) checkAndDeleteStaleConntrackEntries() {
+	namespaces, err := n.watchFactory.GetNamespaces()
+	if err != nil {
+		klog.Errorf("Unable to get pods from informer: %v", err)
+	}
+	for _, namespace := range namespaces {
+		if (len(namespace.Annotations[util.ExternalGatewayPodIPsAnnotation]) > 0) || (len(namespace.Annotations[util.RoutingExternalGWsAnnotation]) > 0) {
```

Comment:
@trozet had another nice idea for improvement. If this node has no local pods on this namespace then we can exit straight away.

Reply:
looking good!
```
$ oc logs -n ovn-kubernetes ovnkube-node-xgrvb -c ovnkube-node | grep SURYA
I0712 19:13:15.011095     881 node.go:678] SURYA [10.244.0.3]
I0712 19:13:49.923939     881 node.go:678] SURYA [10.244.0.3]
I0712 19:14:49.983622     881 node.go:678] SURYA [10.244.0.3]
I0712 19:15:50.054568     881 node.go:678] SURYA [10.244.0.3]
I0712 19:16:50.123751     881 node.go:678] SURYA [10.244.0.3]
$ oc logs -n ovn-kubernetes ovnkube-node-fbt5g -c ovnkube-node | grep SURYA
$ oc logs -n ovn-kubernetes ovnkube-node-wn8xv -c ovnkube-node | grep SURYA
I0712 18:39:28.765682    1883 node.go:678] SURYA [10.244.2.6]
I0712 18:39:36.748368    1883 node.go:678] SURYA [10.244.2.6]
[surya@hidden-temple e2e]$ 

```